### PR TITLE
Migrated notifications to Firebase

### PR DESCRIPTION
**Pull request by eswarasai**
Fixes Issue: https://github.com/snapshot-labs/snapshot-webhook/issues/23

Changes in this PR:
- [x] Removed Pusher Beams integration
- [x] Setup Firebase SDK for receiving push notifications
- [x] Setup service worker to receive background notifications
- [x] Moved notification bell icon from individual space to `Notification` tray on top right corner

How to test and review this PR?
- Configure Firebase credentials (Ref: https://firebase.google.com/docs/cloud-messaging/js/client#configure_web_credentials_with)
- Enable browser notification from the `Notification` tray on top right corner
- On proposal created/deleted event, a push notification is received for the spaces user is following

TODO:
- [ ] Better way to import Firebase config within service worker and `src/helpers/firebase.ts` (Ref: https://github.com/vitejs/vite/discussions/9023)

### CODE REVIEW

**Comment by bonustrack** [comment]
Can you fix the conflicts here?

File: .env [hunk 1]
```diff
@@ -3,5 +3,6 @@ VITE_RELAYER_URL=https://testnet.snapshot.org
 VITE_SCORES_URL=https://score.snapshot.org
 VITE_IPFS_GATEWAY=snapshot.mypinata.cloud
 VITE_DEFAULT_NETWORK=1
-VITE_PUSHER_BEAMS_INSTANCE_ID=2e080021-d495-456d-b2cf-84f9fd718442
-VITE_ENV=develop
\ No newline at end of file
+VITE_ENV=develop
+VITE_VAPID_KEY=
```
Comment (samuveth) [inline .env:7]: These should both have a default value that will work out of the box

File: public/firebase-messaging-sw.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+importScripts(
+  'https://www.gstatic.com/firebasejs/9.10.0/firebase-app-compat.js'
+);
+importScripts(
+  'https://www.gstatic.com/firebasejs/9.10.0/firebase-messaging-compat.js'
+);
+
+const firebaseConfig = {
+  // TODO: config
```
Comment (samuveth) [inline public/firebase-messaging-sw.js:9]: What needs to be done here?

File: public/manifest.json [hunk 3]
```diff
@@ -23,5 +23,6 @@
   "start_url": ".",
   "display": "standalone",
   "theme_color": "#000000",
-  "background_color": "#ffffff"
+  "background_color": "#ffffff",
+  "gcm_sender_id": "XXXXXXXXXXXX"
```
Comment (samuveth) [inline public/manifest.json:27]: What does `XXXXXXXXXXXX` represent?

**Comment by samuveth** [comment]
@eswarasai I've converted this PR to a draft as I see some things missing and the preview deployment doesn't work.

**Comment by samuveth** [comment]
@eswarasai have you tried this https://stackoverflow.com/questions/73355783/proper-way-to-setup-firebase-messaging-service-worker-with-vite

**Comment by samuveth** [comment]
@eswarasai can you resolve conflicts

**Comment by ChaituVR** [comment]
Keep this

**Pull request closed without merging**